### ISSUE TITLE
Iterating over devices works only once

### ISSUE THREAD

**Issue by sol1james** [netbox-community/pynetbox#450]
If I iterate over all devices as shown in the example code in the documentation it works only once.  Ie if I write code like this:

```
>>> devices = nb.dcim.devices.all()
>>> for device in devices:
...     print(device.name)
>>> for device in devices:
...     print(device.name)
```

The print statement in the second for loop doesn't get executed.  If I do another call to all() then I get another iterator but this introduces a noticable pause.  You used to be able to do this without any pauses.

Is this working as designed?

**Comment by zachmoody** [comment]
Yep, see https://github.com/netbox-community/pynetbox/issues/374.

**Comment by sol1james** [comment]
Thanks for that.  I created a [pull request](https://github.com/sol1james/pynetbox/pull/1) to update the documentation to reflect this.

**Comment by abhi1693** [comment]
fixed and merged in #525
